dockers/docker-fpm-frr/docker_init.sh

```diff
@@ -3,20 +3,24 @@
 mkdir -p /etc/frr
 mkdir -p /etc/supervisor/conf.d
 
+ETC_FRR_BACKUP="/etc/frr/backup"
+
+mkdir -p $ETC_FRR_BACKUP
+
 CFGGEN_PARAMS=" \
     -d \
     -y /etc/sonic/constants.yml \
     -t /usr/share/sonic/templates/frr_vars.j2 \
     -t /usr/share/sonic/templates/supervisord/supervisord.conf.j2,/etc/supervisor/conf.d/supervisord.conf \
-    -t /usr/share/sonic/templates/bgpd/gen_bgpd.conf.j2,/etc/frr/bgpd.conf \
+    -t /usr/share/sonic/templates/bgpd/gen_bgpd.conf.j2,$ETC_FRR_BACKUP/bgpd.conf \
     -t /usr/share/sonic/templates/supervisord/critical_processes.j2,/etc/supervisor/critical_processes \
-    -t /usr/share/sonic/templates/zebra/zebra.conf.j2,/etc/frr/zebra.conf \
-    -t /usr/share/sonic/templates/staticd/gen_staticd.conf.j2,/etc/frr/staticd.conf \
-    -t /usr/share/sonic/templates/gen_frr.conf.j2,/etc/frr/frr.conf \
+    -t /usr/share/sonic/templates/zebra/zebra.conf.j2,$ETC_FRR_BACKUP/zebra.conf \
+    -t /usr/share/sonic/templates/staticd/gen_staticd.conf.j2,$ETC_FRR_BACKUP/staticd.conf \
+    -t /usr/share/sonic/templates/gen_frr.conf.j2,$ETC_FRR_BACKUP/frr.conf \
     -t /usr/share/sonic/templates/isolate.j2,/usr/sbin/bgp-isolate \
     -t /usr/share/sonic/templates/unisolate.j2,/usr/sbin/bgp-unisolate \
-    -t /usr/local/sonic/frrcfgd/bfdd.conf.j2,/etc/frr/bfdd.conf \
-    -t /usr/local/sonic/frrcfgd/ospfd.conf.j2,/etc/frr/ospfd.conf \
+    -t /usr/local/sonic/frrcfgd/bfdd.conf.j2,$ETC_FRR_BACKUP/bfdd.conf \
+    -t /usr/local/sonic/frrcfgd/ospfd.conf.j2,$ETC_FRR_BACKUP/ospfd.conf \
```

Comment:
If config type is not `separated`, do these templates need to be rendered?

Reply:
I think they don't need. Should I refactor this script? The current modification try to keep the original code structure without too much refactor.

Reply:
I would recommend doing so since we may generate some files that are never used otherwise. It seems that creating the directory `ETC_FRR_BACKUP` is also unnecessary.

Reply:
Done, please review it.

Reply:
@tahmed-dev It seems that this is a tradeoff between two options:
1. Generating some files that will never be used in some config type (render all templates anyway in one `sonic-cfggen`);
2. Introducing an additional `sonic-cfggen` call in some config type (get the config type in the first call and only render the necessary templates in the second).

I am not completely sure which one would be better. @tahmed-dev What do you think?

Reply:
now there are two sonic-cfggen call. it can increase the fast reboot time. @Pterosaur , can you runa sample call on the device to see how much increase the this additiona sonic-cfggen can take?

Reply:
Once calls
```
root@ARISTA01T1:/# time sonic-cfggen  -d     -y /etc/sonic/constants.yml     -t /usr/share/sonic/templates/frr_vars.j2     -t /usr/share/sonic/templates/supervisord/supervisord.conf.j2     -t /usr/share/sonic/templates/bgpd/gen_bgpd.conf.j2     -t /usr/share/sonic/templates/supervisord/critical_processes.j2     -t /usr/share/sonic/templates/zebra/zebra.conf.j2     -t /usr/share/sonic/templates/staticd/gen_staticd.conf.j2     -t /usr/share/sonic/templates/gen_frr.conf.j2     -t /usr/share/sonic/templates/isolate.j2     -t /usr/share/sonic/templates/unisolate.j2     -t /usr/local/sonic/frrcfgd/bfdd.conf.j2     -t /usr/local/sonic/frrcfgd/ospfd.conf.j2 > /dev/null

real    0m0.267s
user    0m0.179s
sys     0m0.051s
```

Twice call
```
root@ARISTA01T1:/# time bash -c "sonic-cfggen  -d     -y /etc/sonic/constants.yml     -t /usr/share/sonic/templates/frr_vars.j2     -t /usr/share/sonic/templates/supervisord/supervisord.conf.j2     -t /usr/share/sonic/templates/supervisord/critical_processes.j2     -t /usr/share/sonic/templates/isolate.j2     -t /usr/share/sonic/templates/unisolate.j2 && sonic-cfggen -d         -y /etc/sonic/constants.yml         -t /usr/share/sonic/templates/bgpd/gen_bgpd.conf.j2         -t /usr/share/sonic/templates/zebra/zebra.conf.j2         -t /usr/share/sonic/templates/staticd/gen_staticd.conf.j2         -t /usr/local/sonic/frrcfgd/bfdd.conf.j2         -t /usr/local/sonic/frrcfgd/ospfd.conf.j2 " > /dev/null

real    0m0.405s
user    0m0.285s
sys     0m0.087s
```


It increases about 0.2s.

Reply:
@tahmed-dev, any comments?

Reply:
haven't heard back from tamer. let's do it for now.